### PR TITLE
LPS-178557 fix Headless client fails when receiving an HTML

### PR DESCRIPTION
**Pull request by jalber786**
**What's new:**

- Added to HttpResponse class (located in HTTPInvoker.java) a _contentType parameter and it's respective setter and getter.
- Added to client_resource.ftl a condition that checks the response Content-Type to prevent the client from crashing when recieving an HTML response.
- Executed buildREST on /module/apps (**_ NOTE:_** I don't have the permissions to execute de buildREST in dxp/apps becouse I don't have the permissions that allow me to do it)

Jira ticket: https://issues.liferay.com/browse/LPS-178557

CC: @4lejandrito

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
We don't use merge commits. Use rebase instead.

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl [hunk 1]
```diff
@@ -160,7 +160,23 @@ public interface ${schemaName}Resource {
 					_logger.log(Level.WARNING, "HTTP response message: " + httpResponse.getMessage());
 					_logger.log(Level.WARNING, "HTTP response status code: " + httpResponse.getStatusCode());
 
-					throw new Problem.ProblemException(Problem.toDTO(content));
+					Problem.ProblemException problemException = null;
+					String contentType = httpResponse.getContentType();
+
+					if (contentType.equals("text/html;charset=utf-8")){
```
Comment (4lejandrito) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl:166]: Invert condition by checking `application/json`.

Comment (4lejandrito) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl:166]: Invert condition by checking `application/json`.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl [hunk 2]
```diff
@@ -159,7 +160,25 @@ public interface ${schemaName}Resource {
...
+					if (Objects.equals(httpResponse.getContentType(), "application/json")) {
+						problemException = new Problem.ProblemException(
+							Problem.toDTO(content));
+					}
+					else {
+						_logger.log(
+							Level.WARNING,
+							"Unable to process Content-Type: " + httpResponse.getContentType());
+
+						String jsonResponse =
```
Comment (4lejandrito) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl:174]: Avoid generating JSON inline. Instead instantiate a new Problem and set the status.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by jalber786** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:sf

**Review by 4lejandrito — APPROVED** [review]
It's great now!

**Comment by 4lejandrito** [comment]
Issues seem related @jalber786, @sergiojimcos. Please have a look.

**Comment by sergiojimcos** [comment]
@4lejandrito I will take a view at this, but i think that it is necessary to excute buildREST in `modules/dxp/apps`

**Comment by sergiojimcos** [comment]
I will do it! :grin:

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:stop

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
@jalber786 failure seems related (same test you changed?) can you have a look tomorrow?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Tests are failing.

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:stop

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
@sergiojimcos please help @jalber786 run buildREST again.

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward:force

**Comment by magjed4289** [comment]
ci:reevaluate:1347401_3907

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
ci:forward:force

**Comment by sergiojimcos** [comment]
ci:forward:force

**Comment by 4lejandrito** [comment]
Sent manually: https://github.com/brianchandotcom/liferay-portal/pull/133372.

**Comment by 4lejandrito** [comment]
Guys fix conflicts.

**Comment by sergiojimcos** [comment]
Sent to Brian: https://github.com/brianchandotcom/liferay-portal/pull/133374

**Pull request closed without merging**